Modules/Devices/src/Devices.Domain/DomainEvents/Outgoing/IdentityToBeDeletedDomainEvent.cs

```diff
@@ -3,10 +3,12 @@
 namespace Backbone.Modules.Devices.Domain.DomainEvents.Outgoing;
 public class IdentityToBeDeletedDomainEvent : DomainEvent
 {
-    public IdentityToBeDeletedDomainEvent(string identityAddress) : base($"{identityAddress}/IdentityToBeDeleted", randomizeId: true)
+    public IdentityToBeDeletedDomainEvent(string identityAddress, DateTime? gracePeriodEndsAt) : base($"{identityAddress}/IdentityToBeDeleted", randomizeId: true)
     {
         IdentityAddress = identityAddress;
+        GracePeriodEndsAt = gracePeriodEndsAt;
     }
 
     public string IdentityAddress { get; }
+    public DateTime? GracePeriodEndsAt { get; }
```

Comment:
Is there a specific reason (or case) why this has to be null? As far as I understood it, when the event is raised, the Deletion Process is either started by the owner or approved by them (therefore always approved). When approved, the `GracePeriodEndsAt` property is always a valid date (set in the `ApproveInternally` method).

Reply:
In the `IdentityDeletionProcess` file, the `GracePeriodEndsAt` property is declared as nullable, so I maintained that approach to pass it to the domain event. However, your question is valid but unfortunately I don't know the answer.

Reply:
> I don't know the answer.

The answer is easy. If the property CAN be nullable, make it nullable. If it can NOT be nullable, make it not nullable.

And as Mika correctly mentioned, in case of this event the property will never be null.

Reply:
After giving it some thought, I now understand that the only scenario in which the `GracePeriodEndsAt` property is nullable occurs when the deletion process is initiated by support and is still in the approval phase. Changes made.